core/components/atoms/button/button.js

```diff
@@ -170,16 +170,17 @@ const ButtonContent = props => {
 
   let icon = props.success ? 'check' : props.icon
 
+  if (props.text) {
+    content.push(<Button.Text key="text">{props.text}</Button.Text>)
+  }
+
   if (props.loading) {
-    content.push(
-      <Spinner key="spinner" size="small" inverse={getAttributes(props).loadingInverse} />
-    )
+    content.unshift(<Spinner key="spinner" inverse={getAttributes(props).loadingInverse} />)
```

Comment:
This logic is slightly difficult to read

Can we make all these conditionals independent?

```
if (props.text) ...

if (props.loading) ...

if (props.icon && !props.loading) ...
```

Reply:
Yep 👍